echo_test.go

```diff
@@ -161,6 +170,40 @@ func TestEchoStatic(t *testing.T) {
 	}
 }
 
+func TestEchoStaticRedirectIndex(t *testing.T) {
```

Comment:
You should not run tests like that because:
* it adds constant 200ms to execution time - using sleeps makes tests slow in long run and brittle some cases
* it assumes that port 1323 is free - use `127.0.0.1:0` `:0` etc instead. `:0` would bind to random port and you can get the address where server was binded with `addr := e.Listener.Addr()`. See documentation https://golang.org/pkg/net/#Listen
  > If the port in the address parameter is empty or "0", as in "127.0.0.1:" or "[::1]:0", a port number is automatically chosen. The Addr method of Listener can be used to discover the chosen port.

`TestEchoStatic` that you modified above is pretty much same as this is. When you spin up new http server. Each request will invoke `ServeHTTP(rec, req)` method same as `TestEchoStatic` does ... but without starting a server.